Packs/qualys/Integrations/Qualysv2/Qualysv2.py

```diff
@@ -3513,18 +3519,22 @@ def main():  # pragma: no cover
 
             if fetch_stage == 'assets':
 
-                demisto.debug(f'Starting fetch for assets, {start_time=}')
-                assets, new_last_run, total_assets, snapshot_id, set_new_limit = fetch_assets(client=client,
-                                                                                              assets_last_run=last_run)
-                if set_new_limit or check_fetch_duration_time_exceeded(start_time):
+                demisto.debug(f'Starting fetch for assets, {EXECUTION_START_TIME=}')
+                assets, new_last_run, total_assets_to_report, snapshot_id, set_new_limit = fetch_assets(client=client,
+                                                                                                        assets_last_run=last_run)
+                real_amount_of_assets = new_last_run.get("total_assets")
+                if set_new_limit or check_fetch_duration_time_exceeded(EXECUTION_START_TIME):
                     new_last_run = set_last_run_with_new_limit(last_run, last_run.get('limit', HOST_LIMIT))
-                    last_run['nextTrigger'] = '0'
+                    new_last_run['nextTrigger'] = '0'
                 else:
-                    demisto.debug(f'sending {len(assets)} assets to XSIAM. Total assets collected so far: {total_assets}')
+                    demisto.debug(f'sending {len(assets)} assets to XSIAM. Total assets collected so far: '
+                                  f'{real_amount_of_assets}')
                     send_data_to_xsiam(data=assets, vendor=VENDOR, product='assets', data_type='assets',
-                                       snapshot_id=snapshot_id, items_count=str(total_assets), should_update_health_module=False)
+                                       snapshot_id=snapshot_id, items_count=str(total_assets_to_report),
+                                       should_update_health_module=False)
+
                 demisto.setAssetsLastRun(new_last_run)
-                demisto.updateModuleHealth({'{data_type}Pulled'.format(data_type='assets'): total_assets})
+                demisto.updateModuleHealth({'{data_type}Pulled'.format(data_type='assets'): real_amount_of_assets})
```

Comment:
Shouldn't this be part of the 'else'? sunce we did not send any assets in case 'set_new_limit' is true.